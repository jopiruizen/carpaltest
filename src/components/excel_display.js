import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {request_saved_excel,check_connection} from '../actions/store_actions';

//will not dispatch anything but will receive ...


const mapStateToProps = ( state ) => {
	console.log("excel_display.mapStateToProps.updating...");
	console.log(state);
	 return {
	 	excel_content: state.local_store.excel_content,
	 	has_connection: state.local_store.has_connection
	 }
}

const mapDispatchToProps = ( dispatch ) => {
	return bindActionCreators ({request_saved_excel, check_connection} , dispatch);
}


class ExcelDisplay  extends Component {

	checkConnectionDelay = () => {
		this.props.check_connection();
	}

	componentDidMount() {
		this.props.request_saved_excel () ;
 		//this.props.check_connection();
	}	
 
	render() {
		console.log(this.props);
		return  ( 
			<div className="data-content"> 
				{this.renderTable()}
				{this.checkConnectionDelay()}
			</div>
		);
	}

	renderTable = () => {
		if( this.props.excel_content === null || this.props.excel_content === undefined ) {
			return (
				<h2> No Data to Show! </h2>
			);
		} else {
			return ( 
				<table><tbody>
					{this.renderHeader()}
					{this.renderRows()}
				</tbody></table>
			)
		}

		
	}


	renderHeader = () =>  { 
		var header = this.props.excel_content.header;
		return ( 
			<tr>{this.renderHeaderItems( header )}</tr>
		);
 	}

 	renderHeaderItems = (header) =>  {
 		var jsx = [];
 		for ( let i = 0; i < header.length; i++ ) { 			
 			jsx.push( (<th key={i.toString()}>{header[i]  }</th>) );
 		}
 		return jsx;
 	}

	renderRows = () => {
		var rows = this.props.excel_content.rows;

		var jsx = [];
		for( let i =0 ; i < rows.length;i++ ){
			jsx.push (
				(<tr key={i}>{this.renderRowItem(rows[i])}</tr>)
			);
		}
		return jsx;
	}

	renderRowItem = ( item ) => {
		var jsx = [];
		for( let i=0; i < item.length;i++) {
			jsx.push(
				<td key={i}>{item[i]}</td>
			);
		}
		return jsx;
	}
}

export default connect(mapStateToProps, mapDispatchToProps) (ExcelDisplay);