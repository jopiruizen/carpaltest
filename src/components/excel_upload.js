import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';


import {parse_excel} from '../actions/excel_actions'
//will not receive anything.. but will trigger file selection action 


const mapStateToProps = ( state ) => {
	return  ( 
		{
			"somekey": ""
		}	 
	);
}


const mapDispatchToProps = (dispatch) => {
	return bindActionCreators (
		{
			parse_excel
		}, 
		dispatch
	)
}

class ExcelUpload extends Component {

	state = {   "somekey" : "hello" };


 	shouldComponentUpdate ( props, state ) {
 		console.log("Being updated... "   );
 		return true;
 	}

 	_buttonClick = ( event ) => {
 		console.log(this.refs.input_file)
 		this.refs.input_file.click();
 	}

	_inputChange = ( event ) => {
		this.props.parse_excel(event.target);
	}

	render ()  {
		return (	
			<div className="upload-box">
				<input type="file" className="file_btn" ref="input_file"  accept="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"  onChange={this._inputChange }/>
				<input className="upload_btn" type="button" value="UPLOAD EXCEL" onClick={this._buttonClick}/>
			</div>
		)
	}
	 
}

 
export default connect( mapStateToProps, mapDispatchToProps ) (ExcelUpload);


export function  ExcelUpload_instance ()  {
	return <ExcelUpload/>;
}
