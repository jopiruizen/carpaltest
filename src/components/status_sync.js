import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import { check_connection, sync_data } from '../actions/store_actions';

//will dispatch sync from the database to upload files  to the server..
//will show if there is a server connection..


const mapStateToProps = ( state ) => {
	 console.log("Status SYnc: Map State to Props ");
	 console.log(  state );
	 return {
	 	has_connection: state.local_store.has_connection, 
	 	sync_complete: state.local_store.sync_complete
	 }
}


const mapDispatchToProps = ( dispatch ) => {
	return bindActionCreators ({  check_connection ,sync_data } , dispatch);
}


class StatusSync extends Component {

	state = {  button_disabled: true  , is_syncing: false , label:"SYNC" };

	componentDidMount() {
		this.props.check_connection();
	}

	update_syncstatus = () => {
		var me = this;
		function callback( ) {
			me.state.label = "SYNC";
			me.setState({is_syncing: false}  );
		}
		setTimeout(  callback, 5000);
	}
 	
	__syncData = (event) => {
		this.state.is_syncing = true;
 		//this.refs.sync_btn.value = "SYNCING...";
 		this.state.label = "SYNCING...";
 		this.setState( {button_disabled:true});
 		this.props.sync_data();
 		this.update_syncstatus();
	}

	shouldComponentUpdate ( prop, newProp ) {
		/*
		 if( prop.sync_complete == "SYNC_COMPLETE") {
		 	console.log("Sync complete...");		 	 
		 	this.state.is_syncing = false;
		 	console.log("Sync complete..." + this.state.is_syncing );
		 	
		 	return true;
		 }
		*/
		return true;
	}
	 

	 
 
	render() {
		
		console.log("rerender()  " + this.state.is_syncing  + " has COnnection " + this.props.has_connection );

		if(this.props.has_connection === "YES" ) {

			this.state.button_disabled = this.state.is_syncing;		 
		} else {
			this.state.button_disabled = true;
	 	}

		return (
			<div className="sync_status">
				<input ref="sync_btn" type="button" value={this.state.label} disabled={this.state.button_disabled} onClick={this.__syncData} />
			</div>
		);
	}
}

export default connect( mapStateToProps, mapDispatchToProps) (StatusSync);