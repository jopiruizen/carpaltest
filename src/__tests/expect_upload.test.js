
import React from 'react';
import { shallow , mount , configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure ( {adapter: new Adapter()} );




import {ExcelUpload_instance} from '../components/excel_upload';
 
var excelUpload =   shallow (ExcelUpload_instance());

it("ExcelUpload renders correctly.", () => {
	expect ( excelUpload ).toMatchSnapshot();
});

it('Detects excel upload  change ', () => {
 	
 	excelUpload.find(".file_btn").simulate("change");
 	 
}) ; 


it('Fuck this test', () => {


});