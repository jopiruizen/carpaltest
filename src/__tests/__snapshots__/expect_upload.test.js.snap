// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExcelUpload renders correctly. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExcelUpload />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          accept="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"
          className="file_btn"
          onChange={[Function]}
          type="file"
        />,
        <input
          className="upload_btn"
          onClick={[Function]}
          type="button"
          value="UPLOAD EXCEL"
        />,
      ],
      "className": "upload-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel",
          "className": "file_btn",
          "onChange": [Function],
          "type": "file",
        },
        "ref": "input_file",
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "upload_btn",
          "onClick": [Function],
          "type": "button",
          "value": "UPLOAD EXCEL",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            accept="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"
            className="file_btn"
            onChange={[Function]}
            type="file"
          />,
          <input
            className="upload_btn"
            onClick={[Function]}
            type="button"
            value="UPLOAD EXCEL"
          />,
        ],
        "className": "upload-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel",
            "className": "file_btn",
            "onChange": [Function],
            "type": "file",
          },
          "ref": "input_file",
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "upload_btn",
            "onClick": [Function],
            "type": "button",
            "value": "UPLOAD EXCEL",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
